packages/ember-htmlbars/lib/hooks/component.js

```diff
@@ -33,6 +34,29 @@ export default function componentHook(renderNode, env, scope, _tagName, params,
     let componentCell = stream.value();
     if (isComponentCell(componentCell)) {
       tagName = componentCell[COMPONENT_PATH];
+
+      /*
+       * This need to be done to avoid problems with rest positional parameters
+       * when rendered using the dot notation.
+       *
+       * Closure components (for the contextual component feature) does not
+       * actually keep the positional params, but process them at each level.
+       * Therefore, when rendering a closure component with the component
+       * helper we just process the parameters and attributes and then we
+       * merged on top of the closure component attributes.
+       *
+       * In the dot syntax rendering case, it was being doing just by merging
+       * the new attributes with the closure component's and then letting the
+       * rendering process the positional parameters. This caused a problem
+       * with because the new attributes already contain an entry for the rest
+       * positional parameters.
+       *
+       * This is the reason that positional parameters need to be processed
+       * with attributes and then merged with the previous coming from the
+       * closure component.
```

Comment:
And this :-) IMO the first two paragraphs are good enough for code, the rest might be good in the issue description here but that is all.